mesonbuild/build.py

```diff
@@ -1500,7 +1502,7 @@ def check_module_linking(self):
                 return
 
 class Generator(InterpreterHeldObject):
-    def __init__(self, args, kwargs):
+    def __init__(self, args: T.List[TYPE_nvar], kwargs: TYPE_nkwargs) -> None:
```

Comment:
the Generator and Generated List stuff is going to conflict massively when the qt module rework lands.

Reply:
Yeah, but resolving this should be relatively easy (compared to actually creating this PR and making all the tests run again).

I am well aware of the size of this PR and what this means for the review time :)